packages/game-passes/test/GamePasses.test.ts

```diff
@@ -0,0 +1,1251 @@
+      );
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_2)).to.equal(
+        MINT_AMOUNT * 2,
+      );
+
+      // Check treasury received payment
+      expect(await paymentToken.balanceOf(treasury.address)).to.equal(
+        price1 + price2,
+      );
+    });
+
+    it('should not allow minting with expired signature', async function () {
+      const price = ethers.parseEther('0.1');
+      const deadline = (await time.latest()) - 3600; // 1 hour in the past
+      const nonce = 0;
+
+      // Create signature
+      const signature = await createMintSignature(
+        signer,
+        user1.address,
+        TOKEN_ID_1,
+        MINT_AMOUNT,
+        price,
+        deadline,
+        nonce,
+      );
+
+      // Try to mint with expired signature
+      await expect(
+        sandboxPasses
+          .connect(user1)
+          .mint(
+            user1.address,
+            TOKEN_ID_1,
+            MINT_AMOUNT,
+            price,
+            deadline,
+            signature,
+          ),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'SignatureExpired');
+    });
+
+    it('should not allow minting with signature from unauthorized signer', async function () {
+      const price = ethers.parseEther('0.1');
+      const deadline = (await time.latest()) + 3600;
+      const nonce = 0;
+
+      // Create signature from unauthorized user
+      const signature = await createMintSignature(
+        user2, // Not a signer
+        user1.address,
+        TOKEN_ID_1,
+        MINT_AMOUNT,
+        price,
+        deadline,
+        nonce,
+      );
+
+      // Try to mint with invalid signature
+      await expect(
+        sandboxPasses
+          .connect(user1)
+          .mint(
+            user1.address,
+            TOKEN_ID_1,
+            MINT_AMOUNT,
+            price,
+            deadline,
+            signature,
+          ),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'InvalidSigner');
+    });
+
+    it('should not allow exceeding max per wallet', async function () {
+      const price = ethers.parseEther('0.1');
+      const deadline = (await time.latest()) + 3600;
+      const nonce = 0;
+
+      // Approve payment token
+      await paymentToken
+        .connect(user1)
+        .approve(await sandboxPasses.getAddress(), price);
+
+      // Create signature for more than max per wallet
+      const signature = await createMintSignature(
+        signer,
+        user1.address,
+        TOKEN_ID_1,
+        MAX_PER_WALLET + 1,
+        price,
+        deadline,
+        nonce,
+      );
+
+      // Try to mint more than max per wallet
+      await expect(
+        sandboxPasses
+          .connect(user1)
+          .mint(
+            user1.address,
+            TOKEN_ID_1,
+            MAX_PER_WALLET + 1,
+            price,
+            deadline,
+            signature,
+          ),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'ExceedsMaxPerWallet');
+    });
+  });
+
```

Comment:
missing tests for nonce:
- bad nonce
- try more than once

Reply:
Added nonce tests: https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1619/commits/156c1ff5cd776706f8ffe5194258585eda05fc5a